demos/rag_agentic/notebooks/Level5_agentic_MCP_and_RAG.ipynb

```diff
@@ -0,0 +1,669 @@
+    "\n",
+    "client = LlamaStackClient(\n",
+    "    base_url=base_url,\n",
+    "    provider_data={\n",
+    "        \"tavily_search_api_key\": tavily_search_api_key # This is required for demo 2\n",
+    "    }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a85664f8",
+   "metadata": {},
+   "source": [
+    "### Indexing the Documents\n",
+    "- Initialize a new document collection in the target vector DB. All parameters related to the vector DB, such as the embedding model and dimension, must be specified here.\n",
+    "- Provide a list of document URLs to the RAG tool. Llama Stack will handle fetching, conversion and chunking of the documents' content."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "46eebb73",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "5a8f4dc4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from llama_stack_client import RAGDocument\n",
+    "\n",
+    "# define and register the document collection to be used\n",
+    "client.vector_dbs.register(\n",
+    "    vector_db_id=vector_db_id,\n",
+    "    embedding_model=VECTOR_DB_EMBEDDING_MODEL,\n",
+    "    embedding_dimension=VECTOR_DB_EMBEDDING_DIMENSION,\n",
+    "    provider_id=VECTOR_DB_PROVIDER_ID,\n",
+    ")\n",
+    "\n",
+    "# ingest the documents into the newly created document collection\n",
+    "urls = [\n",
+    "    (\"https://docs.redhat.com/en/documentation/openshift_container_platform/4.11/html/support/troubleshooting#troubleshooting-installations.html\", \"application/html\"),\n",
+    "]\n",
+    "documents = [\n",
+    "    RAGDocument(\n",
+    "        document_id=f\"num-{i}\",\n",
+    "        content=url,\n",
+    "        mime_type=url_type,\n",
+    "        metadata={},\n",
+    "    )\n",
+    "    for i, (url, url_type) in enumerate(urls)\n",
+    "]\n",
+    "client.tool_runtime.rag_tool.insert(\n",
+    "    documents=documents,\n",
+    "    vector_db_id=vector_db_id,\n",
+    "    chunk_size_in_tokens=VECTOR_DB_CHUNK_SIZE,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "66044170",
+   "metadata": {},
+   "source": [
+    "### Validate tools are available in our llama-stack instance\n",
+    "\n",
+    "When an instance of llama-stack is redeployed your tools need to re-registered. Also if a tool is already registered with a llama-stack instance, if you try to register one with the same `toolgroup_id`, llama-stack will throw you an error.\n",
+    "\n",
+    "For this reason it is recommended to include some code to validate your tools and toolgroups. This is where the `mcp_url` comes into play. The following code will check that the `builtin::rag`, `builtin::websearch`, `mcp::openshift`, `mcp::custom_tool`  and `mcp::slack` tools are registered as tools, but if the any mcp tool is not listed there, it will attempt to register it using the mcp url.\n",
```

Comment:
small nit, change `but if the any mcp tool is not listed there` to `but if any mcp tool is not listed there`

Reply:
and we aren't using the `mcp::custom_tool` anywhere in this notebook, we should remove that